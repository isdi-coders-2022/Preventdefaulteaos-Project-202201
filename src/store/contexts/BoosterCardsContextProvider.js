import BoosterCardsContext from "./BoosterCardsContext";

const BoosterCardsContextProvider = ({ children }) => {
  return (
    <BoosterCardsContext.Provider value={"hola"}>
      {children}
    </BoosterCardsContext.Provider>
  );
};

export default BoosterCardsContextProvider;
