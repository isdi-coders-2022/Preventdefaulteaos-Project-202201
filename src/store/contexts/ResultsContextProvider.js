import { useReducer } from "react";

import resultsCardsReducer from "../reducers/resultsCardsReducer";
import ResultsContext from "./ResultsContext";

const ResultsContextProvider = ({ children }) => {
  const [resultsCards, dispatch] = useReducer(resultsCardsReducer, []);

  return (
    <ResultsContext.Provider value={{ resultsCards, dispatch }}>
      {children}
    </ResultsContext.Provider>
  );
};

export default ResultsContextProvider;
