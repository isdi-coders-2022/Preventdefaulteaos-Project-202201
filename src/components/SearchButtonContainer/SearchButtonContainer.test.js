import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

import SearchButtonsContainer from "./SearchButtonContainer";
import reactTestRenderer from "react-test-renderer";
import BoosterCardsContextProvider from "../../store/contexts/BoosterCardsContextProvider";
import ResultsContextProvider from "../../store/contexts/ResultsContextProvider";

describe("Given a SearchButtonsContainer component", () => {
  describe("When it recive a SearchButtonsContainer", () => {
    test("Then it should render a button", () => {
      render(
        <BrowserRouter>
          <BoosterCardsContextProvider>
            <ResultsContextProvider>
              <SearchButtonsContainer />
            </ResultsContextProvider>
          </BoosterCardsContextProvider>
        </BrowserRouter>
      );

      const items = screen.getAllByRole("button");

      expect(items).toHaveLength(3);
    });
  });

  describe("when it receive a button with name 'Kamigawa'", () => {
    test("then it should render a button with name 'Kamigawa'", () => {
      render(
        <BrowserRouter>
          <BoosterCardsContextProvider>
            <ResultsContextProvider>
              <SearchButtonsContainer />
            </ResultsContextProvider>
          </BoosterCardsContextProvider>
        </BrowserRouter>
      );
      const text = "Kamigawa";

      const buttonText = screen.getByText(text);

      expect(buttonText).toBeInTheDocument();
    });
  });

  describe("when it recives a SearchButtonsContainer with the text 'Latest expansions:'", () => {
    test("then it render a p with the text 'Latest expansions:'", () => {
      render(
        <BrowserRouter>
          <BoosterCardsContextProvider>
            <ResultsContextProvider>
              <SearchButtonsContainer />
            </ResultsContextProvider>
          </BoosterCardsContextProvider>
        </BrowserRouter>
      );
      const p = "Latest expansions:";

      const pRender = screen.getByText(p);

      expect(pRender).toBeInTheDocument();
    });
  });

  describe("When recive a component", () => {
    test("Then it should render", () => {
      const component = reactTestRenderer.create(
        <BrowserRouter>
          <BoosterCardsContextProvider>
            <ResultsContextProvider>
              <SearchButtonsContainer />
            </ResultsContextProvider>
          </BoosterCardsContextProvider>
        </BrowserRouter>
      );
      expect(component.toJSON()).toMatchSnapshot();
    });
  });
});
