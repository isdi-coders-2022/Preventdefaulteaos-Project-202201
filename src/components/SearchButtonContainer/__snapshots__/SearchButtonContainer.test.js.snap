// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a SearchButtonsContainer component When recive a component Then it should render 1`] = `
<div
  className="sc-bdvvtL kLsyKV searchButtonContainer"
>
  <p>
    Show me the cards
  </p>
  <a
    href="/results"
    onClick={[Function]}
  >
    <button
      className="sc-gsDKAQ sc-eCImPb cvbhQH XiARI"
    >
      Show Cards
    </button>
  </a>
</div>
`;
